src/app/components/kudos-challenge-completed/kudos-challenge-completed.component.js

```diff
@@ -0,0 +1,90 @@
+            $scope.completedChallengesCollection = Resources.getCompletedChallenges();
+            showMoreButton(val);
+        });
+
+        $scope.$watch(function () {
+            return Resources.getCurrentUserEmail()
+        }, function (newVal) {
+            if (!isValid(newVal)) $scope.userEmail = Resources.getCurrentUserEmail();
+        });
+
+        function checkList() {
+            $scope.showList = $scope.completedChallengesCollection.length > 0;
+        }
+
+        function acornPlural(amount) {
+            return amount > 1 ? amount + " Acorns" : amount + " Acorn"
+        }
+
+        $scope.$watch(function () {
+            checkList();
+        });
+
+        function getWinner(index) {
+            if (Resources.getCompletedChallenges()[index].creatorStatus) {
+                if (Resources.getCompletedChallenges()[index].creator == Resources.getCurrentUserEmail) return "You won the challenge!"
+                else return Resources.getCompletedChallenges()[index].creator + " won the challenge";
+            } else if (Resources.getCompletedChallenges()[index].creatorStatus == false){
+                if (Resources.getCompletedChallenges()[index].creator == Resources.getCurrentUserEmail) return Resources.getCompletedChallenges()[index].participant + " won the challenge";
+                else return "You won the challenge!"
+            }
+
+            if (Resources.getCompletedChallenges()[index].participantStatus) {
+                if (Resources.getCompletedChallenges()[index].participant == Resources.getCurrentUserEmail()) return "You won the challenge!"
+                else return Resources.getCompletedChallenges()[index].participant + " won the challenge";
+            } else if (Resources.getCompletedChallenges()[index].participantStatus == false){
+                if (Resources.getCompletedChallenges()[index].participant == Resources.getCurrentUserEmail()) return Resources.getCompletedChallenges()[index].creator + " won the challenge";
+                else return "You won the challenge!"
+            }
+        }
+
+        function showMoreButton(val) {
+            if (val.length > $scope.completedChallengesLimit) {
+                $scope.showMoreButton = true;
+            } else {
+                $scope.showMoreButton = false;
+            }
+        }
+
+        function showMore() {
+            $scope.completedChallengesLimit += showMoreLimit;
+            showMoreButton(Resources.getCompletedChallenges());
+        }
+
+        function showLess() {
+            $scope.completedChallengesLimit = showMoreLimit;
+            showMoreButton(Resources.getCompletedChallenges());
+            $scope.showLessButton = false;
+        }
+
+    };
+
+    CompletedChallengesController.$inject = ['$scope', '$httpParamSerializer', 'CompletedChallengesService', 'Challenges', 'Resources'];
+
+    angular.module('myApp.components.completedChallenges', [])
+        .component('kudosChallengeCompleted', {
```

Comment:
Single responsibility principle - one component per file. I'd have no problem if the controller was declared via function, but here you are declaring a new component.